scripts/us_census/pep_srh/process.py

```diff
@@ -0,0 +1,1015 @@
+                axis=1,
+                inplace=True)
+
+        if file == files_list[0]:
+            df.to_csv(output_file_path + output_file_name, index=False)
+        else:
+            df.to_csv(output_file_path + output_file_name,
+                      index=False,
+                      mode='a')
+
+    # Section 2 - Writing Agg data
+    df = pd.read_csv(output_file_path + output_file_name)
+    output_file_path = _CODEDIR + PROCESS_AGG_DIR + '2010_2020/county/'
+    df1 = pd.DataFrame()
+    df1['YEAR'] = df['YEAR'].copy()
+    df1['LOCATION'] = df['LOCATION'].copy()
+
+    _NOT_HISPANIC_RACES = [
+        'NH', 'NHWA', 'NHBA', 'NHIA', 'NHAA', 'NHNA', 'NHTOM', 'NHWAC', 'NHBAC',
+        'NHIAC', 'NHAAC', 'NHNAC'
+    ]
+    _HISPANIC_RACES = [
+        'H', 'HWA', 'HBA', 'HIA', 'HAA', 'HNA', 'HTOM', 'HWAC', 'HBAC', 'HIAC',
+        'HAAC', 'HNAC'
+    ]
+
+    # df1['NH'] = df1['H'] = 0
+    for p in _NOT_HISPANIC_RACES:
+        df1[p] = (df[p + '_MALE'] + df[p + '_FEMALE']).copy()
+
+    for p in _HISPANIC_RACES:
+        df1[p] = (df[p + '_MALE'] + df[p + '_FEMALE']).copy()
+
+    df1.to_csv(output_file_path + output_file_name, header=True, index=False)
+
+
+def _process_county_files(download_dir):
+    """
+    Process county files from 1990 - 2020
+    Helper function - invokes county_process_file methods for individual decades.
+
+    Args:
+      download_dir: download directory - input files are saved here.
+    """
+    _process_county_files_1990_2000(download_dir)
+    _process_county_files_2000_2010(download_dir)
+    _process_county_files_2010_2020(download_dir)
+
+
+def _consolidate_national_files():
+    """
+    Consolidate all national level files into single national level file.
+    Only SV relevant for SRH processing are retained and all other stats are 
+    dropped.
+
+    This funtion consolidates both as-is and agg data into two seperate files.
+    """
+    national_as_is_files = [
+        _CODEDIR + PROCESS_AS_IS_DIR + yr + '/national/national_' + yr + '.csv'
+        for yr in ['1980_1990', '1990_2000', '2000_2010', '2010_2020']
+    ]
+
+    for file in national_as_is_files:
+        df = pd.read_csv(file)
+
+        # Drop S, SR columns 2000 - 2010 file
+        if file == national_as_is_files[2]:
+            df.drop(_SR_COLUMNS, axis=1, inplace=True)
+
+        # Drop S, SR, Race Combination columns 2010 - 2020 file
+        if file == national_as_is_files[3]:
+            df.drop(_SR_COLUMNS + _SR_CMBN_COLUMNS, axis=1, inplace=True)
+
+        df = df.melt(id_vars=['YEAR', 'LOCATION'],
+                     var_name='SV',
+                     value_name='OBSERVATION')
+        df.replace({"SV": STAT_VAR_COL_MAPPING}, inplace=True)
+        df["SV"] = 'dcid:' + df["SV"]
+
+        df.insert(3, 'MEASUREMENT_METHOD', 'dcs:dcAggregate/CensusPEPSurvey',
+                  True)
+
+        df["MEASUREMENT_METHOD"] = df.apply(
+            lambda r: _calculate_agg_measure_method(r.YEAR, r.SV), axis=1)
```

Comment:
Shouldn't `_calculate_asis_measure_method` be used here or do we always have national files as aggregated?

Reply:
I get it.. asis vs agg is for the SRH vs SH and national is always aggregated from state.